src/Operators/vector_rotation_operators.jl

```diff
@@ -0,0 +1,157 @@
+# TODO: have a general Oceananigans-wide function that retrieves a pointwise
+# value for a function, an array, a number, a field etc?
+# This would be a generalization of `getbc` that could be used everywhere we need it
+@inline getvalue(::Nothing,        i, j, k, grid, args...) = nothing
+@inline getvalue(a::Number,        i, j, k, grid, args...) = a
+@inline getvalue(a::AbstractArray, i, j, k, grid, args...) = @inbounds a[i, j, k]
+
+"""
+    intrinsic_vector(i, j, k, grid::AbstractGrid, uₑ, vₑ, wₑ)
+
+Convert the three-dimensional vector with components `uₑ, vₑ, wₑ` defined in an _extrinsic_ 
+reference frame associated with the domain, to the reference frame _intrinsic_ to the grid. 
+
+_extrinsic_ reference frames are:
+
+- Cartesian for box domains 
```

Comment:
```suggestion
- Cartesian for RectilinearGrid
```

nowhere do we use the term "box domain"?

Reply:
How I see it is that the _extrinsic_ coordinate system is related to the Domain (in this case rectangular, cartesian domains), while the _intrinsic_ coordinate system is related to the grid (aka the discretization). 
For this reason it would be

Reply:
This distinction does not matter at the moment because we only have one `RectlinearGrid`, but it matters conceptually (for example if we have an unstructured grid on a box domain it would still have a cartesian extrinsic reference frame).
I tried to make the explanation more